tests/i18n/tests.py

```diff
@@ -1859,6 +1859,10 @@ def test_page_with_dash(self):
         response = self.client.get('/de-simple-page/')
         self.assertEqual(response.content, b'Yes')
 
+    def test_404_page_no_default_language_redirect(self):
+        response = self.client.get('/non-existent-no-slash', follow=False)
```

Comment:
Yeah that is exactly why I kept it.
Another option is to remove follow and add assert that response.redirect_chain is None, that may be more expressive to what is supposed to be tested.